src/common/pf_bg_worker.c

```diff
@@ -18,13 +18,23 @@
 
 #include <inttypes.h>
 
+#ifdef UNIT_TEST
+/* Disable background worker during unit tests.
+ * Stack reinitialization during test recreates
+ * the backbround task which cause problems.
+ */
+#define BG_WORKER_DISABLE
+#endif
+
 /* Events handled by bg worker task */
 
 #define BG_JOB_EVENT_UPDATE_PORTS_STATUS  BIT (0)
 #define BG_JOB_EVENT_SAVE_ASE_NVM_DATA    BIT (1)
 #define BG_JOB_EVENT_SAVE_IM_NVM_DATA     BIT (2)
 #define BG_JOB_EVENT_SAVE_PDPORT_NVM_DATA BIT (3)
 
+#if !defined(BG_WORKER_DISABLE)
```

Comment:
Kanske bara

#if !defined(UNIT_TEST)
/* Disable background worker during unit tests.
 * Stack reinitialization during test recreates
 * the backbround task which cause problems.*/
 
 Vad tror du?